src/libstore/legacy-ssh-store.cc

```diff
@@ -253,6 +253,10 @@ struct LegacySSHStore : public virtual LegacySSHStoreConfig, public virtual Stor
 
         putBuildSettings(*conn);
 
+        if (GET_PROTOCOL_MINOR(conn->remoteVersion) >= 7) {
+            conn->to << ((int) settings.keepFailed);
+        }
+
```

Comment:
Can’t this go inside `putBuildSettings` ?

Reply:
Good point, updated!